lib/watir-webdriver/locators/element_locator.rb

```diff
@@ -146,6 +159,15 @@ def wd_find_by_regexp_selector(selector, method = :find)
         raise Error, "internal error: unable to build WebDriver selector from #{selector.inspect}"
       end
 
+      if how == :xpath && can_convert_regexp_to_contains?
+        rx_selector.each do |key, value|
+          unless key == :tag_name
```

Comment:
As is, this fails most of our tests due to [the way Watir currently matches regex for text](https://github.com/watir/watir-webdriver/blob/master/lib/watir-webdriver/locators/element_locator.rb#L178).

For instance, our tests often do:
```browser.body(text: /Any text on the page/).present? # ==> true```
where
```browser.body(text: 'Any text on the page').present? # ==> false```
and 
```browser.p(text: 'Any text on the page').present? # ==> true```

This PR forces the text to match the particular tag.

One possible fix is to change this line to: 
```next if key == :tag_name || key == :text``` 

I'll take a look at how much performance increase we get from this change without text, and let you know tomorrow.
I know this would greatly reduce the effectiveness of this PR, but I do like the flexibility (if not the performance) of the current behavior to do things like:
```desired_row = browser.tr(text: /Text in First Column/)```
```desired_row.link(text: 'Edit')```